core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -74,6 +77,8 @@
     private static final Logger logger = LoggerFactory.getLogger(HttpChannelPool.class);
     private static final Channel[] EMPTY_CHANNELS = new Channel[0];
 
+    static final AttributeKey<URI> REQUEST_URI_KEY = AttributeKey.valueOf(URI.class, "REQ_URI");
```

Comment:
- How about just storing `ProxyConfig`, so that we don't need to evaluate the request URI every time we retry a new connection?
- Looks like we have to use `ProxyConfig` as an additional field of `PoolKey`?

Reply:
> we don't need to evaluate the request URI every time we retry a new connection?

In retrospect, I agree and think it makes more sense to evaluate for every request (not only for each connection attempt)
Thanks for pointing this out 👍